substrate/frame/core-fellowship/src/migration.rs

```diff
@@ -68,34 +68,141 @@ mod v0 {
 }
 
-pub struct MigrateToV1<T, I = ()>(PhantomData<(T, I)>);
-impl<T: Config<I>, I: 'static> UncheckedOnRuntimeUpgrade for MigrateToV1<T, I> {
-	#[cfg(feature = "try-runtime")]
-	fn pre_upgrade() -> Result<Vec<u8>, TryRuntimeError> {
-		ensure!(
-			T::MaxRank::get() >= v0::RANK_COUNT as u32,
-			"pallet-core-fellowship: new bound should not truncate"
-		);
-		Ok(Default::default())
+pub mod v1 {
+	use super::*;
+	use frame_system::pallet_prelude::BlockNumberFor as LocalBlockNumberFor;
+
+	pub type MemberStatusOf<T> = MemberStatus<LocalBlockNumberFor<T>>;
+	/// V1 type for [`crate::Member`].
+	#[storage_alias]
+	pub type Member<T: Config<I>, I: 'static> = StorageMap<
+		Pallet<T, I>,
+		Twox64Concat,
+		<T as frame_system::Config>::AccountId,
+		MemberStatusOf<T>,
+		OptionQuery,
+	>;
+
+	pub type ParamsOf<T, I> =
+		ParamsType<<T as Config<I>>::Balance, LocalBlockNumberFor<T>, <T as Config<I>>::MaxRank>;
+	/// V1 type for [`crate::Params`].
+	#[storage_alias]
+	pub type Params<T: Config<I>, I: 'static> =
+		StorageValue<Pallet<T, I>, ParamsOf<T, I>, ValueQuery>;
+
+	pub struct MigrateToV1<T, I = ()>(PhantomData<(T, I)>);
+	impl<T: Config<I>, I: 'static> UncheckedOnRuntimeUpgrade for MigrateToV1<T, I> {
+		#[cfg(feature = "try-runtime")]
+		fn pre_upgrade() -> Result<Vec<u8>, TryRuntimeError> {
+			ensure!(
+				T::MaxRank::get() >= v0::RANK_COUNT as u32,
+				"pallet-core-fellowship: new bound should not truncate"
+			);
+			Ok(Default::default())
+		}
+
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
+			// Read the old value from storage
+			let old_value = v0::Params::<T, I>::take();
+			// Write the new value to storage
+			let new = ParamsOf::<T, I> {
+				active_salary: BoundedVec::defensive_truncate_from(
+					old_value.active_salary.to_vec(),
+				),
+				passive_salary: BoundedVec::defensive_truncate_from(
+					old_value.passive_salary.to_vec(),
+				),
+				demotion_period: BoundedVec::defensive_truncate_from(
+					old_value.demotion_period.to_vec(),
+				),
+				min_promotion_period: BoundedVec::defensive_truncate_from(
+					old_value.min_promotion_period.to_vec(),
+				),
+				offboard_timeout: old_value.offboard_timeout,
+			};
+			Params::<T, I>::put(new);
+			T::DbWeight::get().reads_writes(1, 1)
+		}
+	}
+}
+
+pub mod v2 {
+	use super::*;
+	use crate::BlockNumberFor as NewBlockNumberFor;
+	use frame_system::pallet_prelude::BlockNumberFor as LocalBlockNumberFor;
+
+	/// Converts previous (local) block number into the new one. May just be identity functions
+	/// if sticking with local block number as the provider.
+	pub trait ConvertBlockNumber<L, N> {
+		/// Converts to the new type and finds the equivalent moment in time as relative to the new
+		/// block provider
+		///
+		/// For instance - if your new version uses the relay chain number, you'll want to
+		/// use relay current - ((current local - local) * equivalent_block_duration)
+		fn equivalent_moment_in_time(local: L) -> N;
```

Comment:
As a note, if instead we bound: `L: BlockNumberPovider, N: BlockNumberProvider` and here `(local: L::BlockNumber) -> N::BlockNumber`. Then we can implement this method directly in the trait here.

All we need is the duration conversion.